<?xml version="1.0" encoding="UTF-8"?>
    <source package="java.security">
        <class name="PrivilegedActionException" line="28">
            <extends class="Exception"/>
            <comment line="52">
                use serialVersionUID from JDK 1.2.2 for interoperability                
            </comment>
            <javadoc line="28">
                This exception is thrown by
                  &lt;code&gt;doPrivileged(PrivilegedExceptionAction)&lt;/code&gt; and
                  &lt;code&gt;doPrivileged(PrivilegedExceptionAction,
                  AccessControlContext context)&lt;/code&gt; to indicate
                  that the action being performed threw a checked exception.  The exception
                  thrown by the action can be obtained by calling the
                  &lt;code&gt;getException&lt;/code&gt; method.  In effect, an
                  &lt;code&gt;PrivilegedActionException&lt;/code&gt; is a &quot;wrapper&quot;
                  for an exception thrown by a privileged action.
                  &lt;p&gt;As of release 1.4, this exception has been retrofitted to conform to
                  the general purpose exception-chaining mechanism.  The &quot;exception thrown
                  by the privileged computation&quot; that is provided at construction time and
                  accessed via the {@link #getException()} method is now known as the
                  &lt;i&gt;cause&lt;/i&gt;, and may be accessed via the {@link Throwable#getCause()}method, as well as the aforementioned &quot;legacy method.&quot;                
                <see>
                    PrivilegedExceptionAction                    
                </see>
                <see>
                    AccessController#doPrivileged(PrivilegedExceptionAction)                    
                </see>
                <see>
                    AccessController#doPrivileged(PrivilegedExceptionAction,AccessControlContext)                    
                </see>
            </javadoc>
            <declaration name="serialVersionUID" type="long" line="52"/>
            <declaration name="exception" type="Exception" line="54"/>
            <javadoc line="54">
                @serial                
            </javadoc>
            <javadoc line="59">
                Constructs a new PrivilegedActionException &amp;quot;wrapping&amp;quot;
                  the specific Exception.                
                <param>
                    exception The exception thrown                    
                </param>
            </javadoc>
            <method name="PrivilegedActionException" type="constructor" line="65">
                <params>
                    <param name="exception" type="Exception"/>
                </params>
                <comment line="67">
                    Disallow initCause                    
                </comment>
            </method>
            <javadoc line="70">
                Returns the exception thrown by the privileged computation that
                  resulted in this &lt;code&gt;PrivilegedActionException&lt;/code&gt;.
                  &lt;p&gt;This method predates the general-purpose exception chaining facility.
                  The {@link Throwable#getCause()} method is now the preferred means of
                  obtaining this information.                
                <return>
                    the exception thrown by the privileged computation that
                      resulted in this &lt;code&gt;PrivilegedActionException&lt;/code&gt;.                    
                </return>
                <see>
                    PrivilegedExceptionAction                    
                </see>
                <see>
                    AccessController#doPrivileged(PrivilegedExceptionAction)                    
                </see>
                <see>
                    AccessController#doPrivileged(PrivilegedExceptionAction,AccessControlContext)                    
                </see>
            </javadoc>
            <method name="getException" type="Exception" line="85"/>
            <javadoc line="89">
                Returns the cause of this exception (the exception thrown by
                  the privileged computation that resulted in this
                  &lt;code&gt;PrivilegedActionException&lt;/code&gt;).                
                <return>
                    the cause of this exception.                    
                </return>
                <since>
                    1.4                    
                </since>
            </javadoc>
            <method name="getCause" type="Throwable" line="97"/>
            <method name="toString" type="String" line="101">
                <declaration name="s" type="String" line="102"/>
            </method>
        </class>
    </source>